plugins/woocommerce/src/Blocks/Patterns/PTKPatternsStore.php

```diff
@@ -88,14 +88,17 @@ function () {
 	}
 
 	/**
-	 * Check if the last request was more than one day ago.
+	 * Ensure a recurring fetch patterns action is scheduled.
+	 * This is called by the `action_scheduler_ensure_recurring_actions` hook.
 	 *
-	 * @param int $last_request The last request time.
-	 * @return bool
+	 * @return void
 	 */
-	private function is_older_than_one_day( $last_request ) {
-		$current_time = time();
-		return abs( $last_request - $current_time ) > DAY_IN_SECONDS;
+	public function ensure_recurring_fetch_patterns_if_enabled() {
+		if ( ! $this->allowed_tracking_is_enabled() ) {
+			return;
+		}
+
+		$this->schedule_action_if_not_pending( 'fetch_patterns' );
```

Comment:
I think a constant for `fetch_patterns` might be useful and help avoid any potential typos.